### ISSUE TITLE
use CEs to mimic 1:1 terraform language

### ISSUE THREAD

**Issue by jkone27** [pchalamet/NTerraform#2]
with CEs (computation expressions) it could be possible to write a DSL mapping 1 on 1 on terraform and still be valid F# with intellisense, maybe

**Comment by pchalamet** [comment]
Yes this could help definitively especially for binding trackings!

**Comment by pchalamet** [comment]
But probably as you know, implementing a lang on top of terraform is probably vain without dealing to plugins communication and coordination (the tf syntax tree is used for both parsing and DAG execution) . This is basically what has done Pulumi and this is why I think this POC was funny but can't go much far without big investment.
